### DISCUSSION TITLE
Page customization

### DISCUSSION THREAD

**Discussion by matt-shirey** [Z3d0X/filament-fabricator#80] [Q&A] [answered]
Still pretty new to Fabricator and I haven't found any documentation for this but wanted to see if anyone had a way of doing it.

- Is there a way to add SEO fields as a sidebar?
- Is there a way to limit blocks to specific templates?

**Comment by matt-shirey** [comment]
Okay, I found the visible method that I can invoke on my blocks to only show specific one's based on the template.

I also made a SEO block but I'm trying to find out if there is a way that I can automatically add that block to specific templates as the first item and it can't be deleted.

**Comment by Z3d0X** [comment, marked as answer]
To add SEO fields to sidebar you can use schema slots. You can place this in a service provider
```php
FilamentFabricator::registerSchemaSlot('sidebar.after', [
    TextInput::make('seo_title'),
    //Other fields
]);
```

If you need further customization (other than simply adding fields), you could also override the "PageResource" from the config

**Reply by aasuncion-fol** [reply]
While doing this in register() AppServiceProvider. I encountered this issue

Class "Z3d0X\FilamentFabricator\FilamentFabricator" not found

**Reply by Z3d0X** [reply]
Try using it in the "boot()" of the service provider
